pkg/aggregator/apis/aggregation/v0alpha1/types.go

```diff
@@ -0,0 +1,136 @@
+package v0alpha1
+
+import (
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type DataPlaneService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataPlaneServiceSpec   `json:"spec,omitempty"`
+	Status DataPlaneServiceStatus `json:"status,omitempty"`
+}
+
+type DataPlaneServiceSpec struct {
+	PluginID   string     `json:"pluginID"`
+	PluginType PluginType `json:"pluginType"`
+	Group      string     `json:"group"`
+	Version    string     `json:"version"`
+	// Services is a list of services that the proxied service provides.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Services []Service `json:"services"`
+}
+
+// Service defines the type of service the proxied service provides.
+type Service struct {
+	// Type is the type of service to proxy.
+	Type ServiceType `json:"type"`
+	// Method is the HTTP method to use when proxying the service.
+	// +optional
+	Method string `json:"method,omitempty"`
+	// Path is used by the CustomRouteServiceType and SubResourceServiceType to specify the path to the endpoint.
+	// +optional
+	Path string `json:"path,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type DataPlaneServiceList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []DataPlaneService `json:"items,omitempty"`
+}
+
+// PluginType defines the type of plugin backing the proxied service.
+// +enum
+type PluginType string
+
+// PluginType values
+const (
+	AppPluginType        PluginType = "app"
+	DataSourcePluginType PluginType = "datasource"
+)
+
+// ServiceType defines the type of services the proxied service provides.
+// +enum
+type ServiceType string
+
+// ServiceType values
+const (
+	// AdmissionControlServiceType maps to pluginv2.AdmissionControl
+	AdmissionControlServiceType ServiceType = "admission"
+	// ConversionServiceType maps to pluginv2.ResourceConversion
+	ConversionServiceType ServiceType = "conversion"
+	// DataSourceServiceType maps to pluginv2.Data
+	DataServiceType ServiceType = "data"
```

Comment:
I don't think we want to expose this with anything other than query

Reply:
i was matching the name of the pluginv2 service. i'll switch it to query